src/main/scala/com/github/unchama/seichiassist/subsystems/gacha/System.scala

```diff
@@ -0,0 +1,107 @@
+  BukkitLotteryOfGachaItems
+}
+import com.github.unchama.seichiassist.subsystems.gacha.bukkit.command.GachaCommand
+import com.github.unchama.seichiassist.subsystems.gacha.bukkit.listeners.PlayerPullGachaListener
+import com.github.unchama.seichiassist.subsystems.gacha.domain.{
+  CanBeSignedAsGachaPrize,
+  GachaPrize,
+  GachaPrizeId,
+  GachaPrizeListPersistence
+}
+import com.github.unchama.seichiassist.subsystems.gacha.infrastructure.JdbcGachaPrizeListPersistence
+import com.github.unchama.seichiassist.subsystems.gacha.subsystems.gachaticket.domain.GachaTicketFromAdminTeamRepository
+import com.github.unchama.seichiassist.subsystems.gacha.subsystems.gachaticket.infrastructure.JdbcGachaTicketFromAdminTeamRepository
+import org.bukkit.command.TabExecutor
+import org.bukkit.entity.Player
+import org.bukkit.event.Listener
+import org.bukkit.inventory.ItemStack
+
+trait System[F[_]] extends Subsystem[F] {
+  val api: GachaAPI[F, ItemStack, Player]
+}
+
+object System {
+
+  import cats.implicits._
+
+  def wired[F[_]: OnMinecraftServerThread: NonServerThreadContextShift: ConcurrentEffect]
+    : F[System[F]] = {
+    implicit val serializeAndDeserialize: SerializeAndDeserialize[Nothing, ItemStack] =
+      BukkitItemStackSerializeAndDeserialize
+    implicit val gachaPersistence: GachaPrizeListPersistence[F, ItemStack] =
+      new JdbcGachaPrizeListPersistence[F, ItemStack]()
+    implicit val gachaTicketPersistence: GachaTicketFromAdminTeamRepository[F] =
+      new JdbcGachaTicketFromAdminTeamRepository[F]
+    implicit val canBeSignedAsGachaPrize: CanBeSignedAsGachaPrize[ItemStack] =
+      BukkitCanBeSignedAsGachaPrize
+    implicit val lotteryOfGachaItems: BukkitLotteryOfGachaItems[F] =
+      new BukkitLotteryOfGachaItems[F]
+
+    val system: System[F] = new System[F] {
+      override implicit val api: GachaAPI[F, ItemStack, Player] =
+        new GachaAPI[F, ItemStack, Player] {
+
+          override protected implicit val F: Functor[F] = implicitly
+
+          override def load: F[Unit] = gachaPersistence.list.flatMap { gachaPrizes =>
+            gachaPrizesListRepository.set(gachaPrizes)
+          }
+
+          override def replace(gachaPrizesList: Vector[GachaPrize[ItemStack]]): F[Unit] =
+            gachaPrizesListRepository.set(gachaPrizesList)
+
+          override def removeByGachaPrizeId(gachaPrizeId: GachaPrizeId): F[Unit] = for {
+            prizes <- list
+            targetPrize = prizes.filter(_.id == gachaPrizeId)
+            _ <- replace(prizes.diff(targetPrize))
+          } yield ()
+
+          override def addGachaPrize(gachaPrize: GachaPrizeByGachaPrizeId): F[Unit] =
+            for {
+              prizes <- list
+              newList = gachaPrize(
+                GachaPrizeId(if (prizes.nonEmpty) prizes.map(_.id.id).max + 1 else 1)
+              ) +: prizes
+
+              _ <- replace(newList)
+            } yield ()
+
+          protected implicit val gachaPrizesListRepository
```

Comment:
```suggestion
          protected implicit val gachaPrizesListReference
```
複数のデータを保存するデータストアではないので `Repository` より `Reference` のほうがわかりやすいかも